File: 2-Regression/6-Random_Forest_Regression/src/random_forest_regression.py
# Random Forest Regression

# Import the Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import os

from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestRegressor

# Import the data
Path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'bin\\Position_Salaries.csv'))
BinPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'bin'))
dataset = pd.read_csv(Path)
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2:].values

# Fitting a new result
regressor = RandomForestRegressor(n_estimators=1000,random_state=0)
regressor.fit(X,Y)

# Predicting a new result
Y_pred = regressor.predict(np.array([[6.5]]))
print(Y_pred)

# Visualising the SVR result
RFR = plt.figure(1)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, Y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.scatter([6.5], Y_pred, color = 'green')
plt.title('Truth of Bluff (Random Forest Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

RFR.savefig(BinPath + '\RandomForestRegression_1000_Trees_PY.png')

File: 2-Regression/6-Random_Forest_Regression/src/random_forest_regression.R
# Randon Forest Regression

# Path to the file and Libraries
Path = 'MachineLearning-Portfolio/2-Regression/6-Random_Forest_Regression/bin/Position_Salaries.csv'
BinPath = 'MachineLearning-Portfolio/2-Regression/6-Random_Forest_Regression/bin'
#install.packages('caTools', repos = "http://cran.us.r-project.org", dependencies = TRUE)
#install.packages('ggplot2', repos = "http://cran.us.r-project.org", dependencies = TRUE)
#install.packages('randomForest', repos = "http://cran.us.r-project.org", dependencies = TRUE)

library(caTools, help, pos = 2, lib.loc = NULL)
library(ggplot2)
library(randomForest)

# Import dataset
dataset = read.csv(Path)
dataset = dataset[2:3]

# Fittin the SVR to the dataset
set.seed(1234)
regressor = randomForest(x= dataset[1],
                        y = dataset$Salary,
                        ntree = 300)
# dataset[2] returns a dataframe
# dataset$Salary returns a vector

# Predicting a new result
Y_pred = predict(regressor, data.frame(Level = 6.5))

# Visualsing the SVR result
X_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
SVR <- ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary),
		color = 'red') +
	geom_point(aes(x = 6.5, y = Y_pred),
		color = 'black') +
	geom_line(aes(x = X_grid, y = predict(regressor, newdata = data.frame(Level = X_grid))),
		color = 'blue') +
	ggtitle('Truth of Bluff (Randon Forest Regression)') +
	xlab('Level') +
	ylab('Salary')

ggsave("RandomForestRegression_R.png",  path = BinPath)